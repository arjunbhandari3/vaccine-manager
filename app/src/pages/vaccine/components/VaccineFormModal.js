import moment from "moment";
import { Modal, Form } from "antd";
import React, { useEffect, useState } from "react";

import VaccineForm from "./VaccineForm";

const VaccineFormModal = ({ vaccine, open, onCancel }) => {
  const [form] = Form.useForm();

  const [isSubmitting, setIsSubmitting] = useState(false);

  useEffect(() => {
    if (vaccine) {
      form.setFieldsValue({
        ...vaccine,
        releaseDate: moment(vaccine.releaseDate, "YYYY-MM-DD"),
        expirationDate: moment(vaccine.expirationDate, "YYYY-MM-DD"),
      });
    }
  }, [vaccine, form]);

  return (
    <Modal
      open={open}
      onCancel={onCancel}
      title={
        <h1 className="vaccines-header-title">
          {vaccine ? "Edit vaccine" : "Add vaccine"}
        </h1>
      }
      onOk={() => form.submit()}
      okText={vaccine ? "Update Vaccine" : "Add Vaccine"}
      confirmLoading={isSubmitting}
    >
      <VaccineForm
        form={form}
        vaccine={vaccine}
        onClose={onCancel}
        setIsSubmitting={setIsSubmitting}
      />
    </Modal>
  );
};

export default VaccineFormModal;
