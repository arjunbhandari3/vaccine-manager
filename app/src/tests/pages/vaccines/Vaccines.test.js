import React from "react";
import { setupServer } from "msw/node";
import { act } from "react-dom/test-utils";
import {
  screen,
  render,
  cleanup,
  waitFor,
  fireEvent,
  waitForElementToBeRemoved,
} from "tests/render";

import Vaccines from "pages/vaccine/Vaccines";
import VaccineFormModal from "pages/vaccine/components/VaccineFormModal";
import {
  vaccineResponse,
  allVaccinesResponse,
  vaccineCountResponse,
} from "tests/handlers";

jest.mock(
  "antd-img-crop",
  () =>
    ({ children }) =>
      children
);

// setup server
const server = setupServer(
  allVaccinesResponse,
  vaccineCountResponse,
  vaccineResponse
);

describe("Vaccines", () => {
  beforeAll(() => server.listen());

  afterEach(() => {
    server.resetHandlers();
    cleanup();
  });

  afterAll(() => server.close());

  it('should render "No data" when there is no vaccine', async () => {
    render(<Vaccines />);

    expect(screen.getByText("No data")).toBeInTheDocument();
  });

  it("should render", async () => {
    render(<Vaccines />);

    expect(screen.getByText("All Vaccines")).toBeInTheDocument();
    expect(screen.getByText("No data")).toBeInTheDocument();

    await waitForElementToBeRemoved(screen.queryByText("No data"));

    await waitFor(
      () => screen.getByText("Vaccine 1") && screen.getByText("Vaccine 2")
    );
  });

  it("should render vaccine count", async () => {
    await render(<Vaccines />);

    expect(await screen.getByTestId("total-count").textContent).toBe(
      "Total: 2"
    );
    expect(await screen.getByTestId("mandatory-count").textContent).toBe(
      "Mandatory: 1"
    );
    expect(await screen.getByTestId("optional-count").textContent).not.toBe(
      "Optional: 2"
    );
  });

  it("should highlight mandatory tab when mandatory tab is clicked", async () => {
    await render(<Vaccines />);

    const mandatoryButton = screen.getByTestId("mandatory-count");
    expect(mandatoryButton.textContent).toBe("Mandatory: 1");
    act(() => mandatoryButton.click());
    expect(
      mandatoryButton.className.includes("ant-tag-has-color")
    ).toBeTruthy(); // check if the tag is highlighted
  });

  it('should search for "Vaccine 2" when "Vaccine 2" is typed in search input', async () => {
    await render(<Vaccines />);
    await waitForElementToBeRemoved(screen.queryByText("No data"));

    const searchInput = screen.getByPlaceholderText("Search Vaccines");
    fireEvent.change(searchInput, { target: { value: "Vaccine 2" } });

    //press enter
    fireEvent.keyDown(searchInput, { key: "Enter", code: "Enter" });
    await screen.findByText("Vaccine 2");

    await expect(screen.getByText("Vaccine 2")).toBeInTheDocument();
  });

  it('should open add vaccine modal when "Add Vaccine" button is clicked', async () => {
    await render(<Vaccines />);

    const addVaccineButton = screen.getByRole("button", {
      name: "Add vaccine",
    });
    act(() => addVaccineButton.click());

    //open modal and check if the modal is open
    await render(<VaccineFormModal />);

    expect(screen.getByRole("dialog")).toBeInTheDocument();
    expect(
      screen.getByRole("heading", { name: "Add vaccine" })
    ).toBeInTheDocument();
  });

  it('should open edit vaccine modal when "Edit" button is clicked and should have the correct data', async () => {
    await render(<Vaccines />);
    await waitForElementToBeRemoved(screen.queryByText("No data"));

    //search by data-icon="edit" and click the button
    const editVaccineButton = screen.getByTestId("edit-vaccine-1");
    act(() => editVaccineButton.click());

    //open modal and check if the modal is open
    expect(screen.getByRole("dialog")).toBeInTheDocument();
    expect(
      screen.getByRole("heading", { name: "Edit vaccine" })
    ).toBeInTheDocument();

    //check if the form is filled with the correct data
    expect(screen.getByRole("textbox", { name: "Name" }).value).toBe(
      "Vaccine 1"
    );
    expect(screen.getByRole("textbox", { name: "Description" }).value).toBe(
      "vaccine 1 desc"
    );
  });

  it("should be able to edit and save vaccine", async () => {
    await render(<Vaccines />);
    await waitForElementToBeRemoved(screen.queryByText("No data"));

    //search by data-icon="edit" and click the button
    const editVaccineButton = screen.getByTestId("edit-vaccine-1");
    act(() => editVaccineButton.click());

    //open modal and check if the modal is open
    expect(screen.getByRole("dialog")).toBeInTheDocument();

    //edit the form
    const descriptionInput = screen.getByRole("textbox", {
      name: "Description",
    });
    fireEvent.change(descriptionInput, {
      target: { value: "vaccine 1 description" },
    });

    const releaseDateInput = screen.getByRole("textbox", {
      name: "Release Date",
    });
    fireEvent.change(releaseDateInput, { target: { value: "2021-01-01" } });

    const mandatoryCheckbox = screen.getByRole("switch", {
      name: "Is Mandatory",
    });
    fireEvent.click(mandatoryCheckbox);

    const photoUrl = screen.getByTestId("upload");
    fireEvent.change(photoUrl, { target: { value: "photoUrl.png" } });

    const saveButton = screen.getByRole("button", { name: "Update Vaccine" });

    //save the form
    act(() => saveButton.click());

    //check if the form is saved
    await screen.findByText("Vaccine 1");
    expect(screen.getByText("vaccine 1 description")).toBeInTheDocument();
  });

  it('should open delete vaccine modal when "Delete" button is clicked', async () => {
    await render(<Vaccines />);
    await waitForElementToBeRemoved(screen.queryByText("No data"));

    const deleteVaccineButton = screen.getByTestId("delete-vaccine-1");
    act(() => deleteVaccineButton.click());

    await screen.findByRole("dialog");

    expect(screen.getByRole("dialog")).toBeInTheDocument();

    expect(screen.getByText("Delete Vaccine")).toBeInTheDocument();
    expect(
      screen.getByText("Are you sure you want to delete this vaccine?")
    ).toBeInTheDocument();
  });

  //snapshot
  it("should match snapshot", async () => {
    const { asFragment } = await render(<Vaccines />);

    // await waitForElementToBeRemoved(screen.queryByText("No data"));

    expect(await asFragment()).toMatchSnapshot();
  });
});
