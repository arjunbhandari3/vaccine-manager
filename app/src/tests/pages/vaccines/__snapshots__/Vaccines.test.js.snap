// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vaccines should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="vaccines-container"
  >
    <div
      class="ant-page-header ant-page-header-ghost"
    >
      <div
        class="ant-page-header-heading"
      >
        <div
          class="ant-page-header-heading-left"
        >
          <span
            class="ant-page-header-heading-title"
          >
            <h1
              class="vaccines-header-title"
            >
              All Vaccines
            </h1>
          </span>
          <span
            class="ant-page-header-heading-tags"
          >
            <span
              class="ant-tag ant-tag-has-color"
              data-testid="total-count"
              style="background-color: rgb(24, 144, 255); cursor: pointer;"
            >
              Total: 2
            </span>
            <span
              class="ant-tag ant-tag-red"
              data-testid="mandatory-count"
              style="cursor: pointer;"
            >
              Mandatory: 1
            </span>
            <span
              class="ant-tag ant-tag-green"
              data-testid="optional-count"
              style="cursor: pointer;"
            >
              Optional: 1
            </span>
          </span>
        </div>
        <span
          class="ant-page-header-heading-extra"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
              style="margin-right: 8px;"
            >
              <button
                class="ant-btn ant-btn-primary"
                type="button"
              >
                <span>
                  Add vaccine
                </span>
              </button>
            </div>
            <div
              class="ant-space-item"
            >
              <span
                class="ant-input-affix-wrapper"
              >
                <input
                  class="ant-input"
                  placeholder="Search Vaccines"
                  type="text"
                  value=""
                />
                <span
                  class="ant-input-suffix"
                >
                  <span
                    class="ant-input-clear-icon ant-input-clear-icon-hidden"
                    role="button"
                    tabindex="-1"
                  >
                    <span
                      aria-label="close-circle"
                      class="anticon anticon-close-circle"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="close-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </span>
      </div>
    </div>
    <div>
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: fixed;"
                  >
                    <colgroup>
                      <col
                        style="width: 80px;"
                      />
                      <col
                        style="width: 80px;"
                      />
                      <col />
                      <col
                        style="width: 120px;"
                      />
                      <col
                        style="width: 200px;"
                      />
                      <col
                        style="width: 150px;"
                      />
                      <col />
                      <col />
                      <col />
                      <col
                        style="width: 200px;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          S.N.
                        </th>
                        <th
                          class="ant-table-cell"
                          style="text-align: center;"
                        />
                        <th
                          class="ant-table-cell"
                        >
                          Name
                        </th>
                        <th
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          Mandatory
                        </th>
                        <th
                          class="ant-table-cell ant-table-cell-ellipsis"
                          title="Description"
                        >
                          Description
                        </th>
                        <th
                          class="ant-table-cell ant-table-cell-ellipsis"
                          title="Manufacturer"
                        >
                          Manufacturer
                        </th>
                        <th
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          No. of Doses
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Release Date
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Expiration Date
                        </th>
                        <th
                          class="ant-table-cell ant-table-cell-ellipsis"
                        >
                          <div
                            class="d-flex align-items-center"
                          >
                            <div>
                              Allergies
                            </div>
                            <div
                              class="table-header-icon"
                            >
                              <span
                                aria-label="info-circle"
                                class="anticon anticon-info-circle ml-2 cursor-pointer"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="info-circle"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                                  />
                                  <path
                                    d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                                  />
                                </svg>
                              </span>
                            </div>
                          </div>
                        </th>
                        <th
                          class="ant-table-cell"
                        />
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          1
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <span
                            class="ant-avatar ant-avatar-circle ant-avatar-image"
                            style="width: 40px; height: 40px; line-height: 40px; font-size: 18px;"
                          >
                            <img
                              src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSPtjpM7aoR4kbaH9c6j-uKN4SDiuiekzLTgA&usqp=CAU"
                            />
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          Vaccine 1
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <div
                            class="cursor-pointer"
                          >
                            <span
                              aria-label="star"
                              class="anticon anticon-star"
                              role="img"
                              style="color: orange; font-size: 20px;"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="star"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                                />
                              </svg>
                            </span>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell ant-table-cell-ellipsis"
                        >
                          <span>
                            vaccine 1 desc
                          </span>
                        </td>
                        <td
                          class="ant-table-cell ant-table-cell-ellipsis"
                        >
                          <span>
                            ABC
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          2
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          2022-11-08
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          2022-11-29
                        </td>
                        <td
                          class="ant-table-cell ant-table-cell-ellipsis"
                        >
                          <div
                            class="d-flex flex-wrap"
                          >
                            <span
                              class="ant-tag ant-tag-has-color mb-10x"
                              style="background-color: rgb(0, 200, 81);"
                            >
                              <span>
                                Allergy 1
                              </span>
                            </span>
                            <span
                              class="ant-tag ant-tag-has-color mb-10x"
                              style="background-color: rgb(214, 39, 40);"
                            >
                              <span>
                                Allergy 2
                              </span>
                            </span>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-space ant-space-horizontal ant-space-align-center actions-container"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 8px;"
                            >
                              <div
                                class="cursor-pointer"
                                data-testid="edit-vaccine-1"
                              >
                                <span
                                  aria-label="edit"
                                  class="anticon anticon-edit"
                                  role="img"
                                  style="font-size: 20px; color: blue;"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="edit"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                                    />
                                  </svg>
                                </span>
                              </div>
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <div
                                class="cursor-pointer"
                                data-testid="delete-vaccine-1"
                              >
                                <span
                                  aria-label="delete"
                                  class="anticon anticon-delete"
                                  role="img"
                                  style="font-size: 20px; color: red;"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="delete"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                                    />
                                  </svg>
                                </span>
                              </div>
                            </div>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="2"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          2
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <span
                            class="ant-avatar ant-avatar-circle ant-avatar-image"
                            style="width: 40px; height: 40px; line-height: 40px; font-size: 18px;"
                          >
                            <img
                              src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSPtjpM7aoR4kbaH9c6j-uKN4SDiuiekzLTgA&usqp=CAU"
                            />
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          Vaccine 2
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <div
                            class="cursor-pointer"
                          >
                            <span
                              aria-label="star"
                              class="anticon anticon-star"
                              role="img"
                              style="font-size: 20px;"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="star"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                                />
                              </svg>
                            </span>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell ant-table-cell-ellipsis"
                        >
                          <span>
                            vaccine 2 desc
                          </span>
                        </td>
                        <td
                          class="ant-table-cell ant-table-cell-ellipsis"
                        >
                          <span>
                            ABC
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          3
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          2022-11-10
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          2022-11-29
                        </td>
                        <td
                          class="ant-table-cell ant-table-cell-ellipsis"
                        >
                          <div
                            class="d-flex flex-wrap"
                          >
                            <span
                              class="ant-tag ant-tag-has-color mb-10x"
                              style="background-color: rgb(240, 173, 78);"
                            >
                              <span>
                                Allergy 3
                              </span>
                            </span>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-space ant-space-horizontal ant-space-align-center actions-container"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 8px;"
                            >
                              <div
                                class="cursor-pointer"
                                data-testid="edit-vaccine-2"
                              >
                                <span
                                  aria-label="edit"
                                  class="anticon anticon-edit"
                                  role="img"
                                  style="font-size: 20px; color: blue;"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="edit"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                                    />
                                  </svg>
                                </span>
                              </div>
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <div
                                class="cursor-pointer"
                                data-testid="delete-vaccine-2"
                              >
                                <span
                                  aria-label="delete"
                                  class="anticon anticon-delete"
                                  role="img"
                                  style="font-size: 20px; color: red;"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="delete"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                                    />
                                  </svg>
                                </span>
                              </div>
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
            <ul
              class="ant-pagination ant-table-pagination ant-table-pagination-right"
            >
              <li
                aria-disabled="true"
                class="ant-pagination-prev ant-pagination-disabled"
                title="Previous Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="left"
                    class="anticon anticon-left"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a
                  rel="nofollow"
                >
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-next ant-pagination-disabled"
                title="Next Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="right"
                    class="anticon anticon-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="ant-pagination-options"
              >
                <div
                  aria-label="Page Size"
                  class="ant-select ant-pagination-options-size-changer ant-select-single ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="undefined_list_0"
                        aria-autocomplete="list"
                        aria-controls="undefined_list"
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-label="Page Size"
                        aria-owns="undefined_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="10 / page"
                    >
                      10 / page
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
